### PR TITLE
Share common logic in indentation.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @heejaechang

**Comment by CyrusNajmabadi** [comment]
retest windows_release_unit64_prtest please

**Comment by CyrusNajmabadi** [comment]
Tagging @mavasani as well.

**Comment by CyrusNajmabadi** [comment]
test windows_release_unit64_prtest please

**Comment by CyrusNajmabadi** [comment]
test linux_debug_prtest please

File: src/EditorFeatures/Core/Implementation/SmartIndent/AbstractIndentationService.AbstractIndenter.cs [hunk 1]
```diff
@@ -57,27 +59,69 @@ internal abstract class AbstractIndenter
...
+                    return IndentFromStartOfLine(0);
+                }
+
+                var previousNonWhitespaceOrPreprocessorLine = previousLineOpt.Value;
+
+                // If the user wants block indentation, then we just return the indentation
+                // of the last piece of real code.  
+                //
+                // TODO(cyrusn): It's not clear to me that this is correct.  Block indentation
+                // should probably follow the indentation of hte last non-blank line *regardless
```
Comment (DustinCampbell) [inline src/EditorFeatures/Core/Implementation/SmartIndent/AbstractIndentationService.AbstractIndenter.cs]: hte -> the :trollface:

**Pull request merged**